### PR TITLE
[Blocks] Fix converting list items

### PR DESCRIPTION
**Pull request by remidej**
### What does it do?

Fix the conversion of list items.

Before, when selecting a different type of block while the selection was in a list, it created the block inside of the list. This created UI and validation errors, because only list items and lists are allowed as children of a list.

Now, the parent list will be split in 2, and the modified block will sit in between the 2 list blocks.

### How to test it?

Convert a list item to another block type, save and publish, you should not get a validation error

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js [hunk 1]
```diff
@@ -362,4 +362,96 @@ describe('BlocksEditor toolbar', () => {
     // The dropdown should show only one option selected which is the block content in the first row
     expect(within(headingsDropdown).getByText(/heading 1/i)).toBeInTheDocument();
   });
+
+  it('slits the parent list when converting a list item to another type', async () => {
```
Comment (simotae14) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js:422]: typo
`it('slits the parent list when converting a list item to another type', async () => {`

**Review by simotae14 — APPROVED** [review]
Fix the typo, overall it works for me
The only thing strange but not related to this bugfix when you convert back the block item to a list item there is too much empty space between one line and the other as you can see in this picture
space

**Comment by remidej** [comment]
>The only thing strange but not related to this bugfix when you convert back the block item to a list item there is too much empty space between one line and the other as you can see in this picture

@simotae14 I think when converting a block like a paragraph to a list, we always want to create a new list, regardless of whether the sibling blocks are lists. I agree that it creates a weird blank space, but that seems to just be because lists have big margins (16px top and bottom), which is very visible when you have 2 in a row.

**Review by simotae14 — DISMISSED** [review]
it works and looks good to me

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js [hunk 2]
```diff
@@ -418,4 +418,96 @@ describe('BlocksEditor toolbar', () => {
...
+      {
+        type: 'list',
+        format: 'ordered',
+        children: [
+          {
+            type: 'list-item',
+            children: [
+              {
+                type: 'text',
+                text: 'A line of text in a paragraph.',
```
Comment (madhurisandbhor) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/tests/index.test.js:433]: having varied text would help to read the test case easily, like first item , second item...etc.

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js [hunk 3]
```diff
@@ -151,6 +151,12 @@ const toggleBlock = (editor, value) => {
   };
 
   if (editor.selection) {
+    // If the selection is inside a list, split the list so that the modified block is outside of it
+    Transforms.unwrapNodes(editor, {
+      match: (node) => node.type === 'list',
+      split: true,
+    });
```
Comment (madhurisandbhor) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js:158]: didn't test but that was easy 👍 👍

**Comment by madhurisandbhor** [comment]
also unit tests check is failing, please check

**Review by simotae14 — APPROVED** [review]

**Pull request merged**